tests/simple_server.py

```diff
@@ -25,40 +25,42 @@
     http://docs.python.org/library/simplehttpserver.html#module-SimpleHTTPServer
 """
 
-import sys
-import random
 import socketserver
+import sys
 from http.server import SimpleHTTPRequestHandler
+from typing import Type, Union
 
 
 class QuietHTTPRequestHandler(SimpleHTTPRequestHandler):
-  """A SimpleHTTPRequestHandler that does not write incoming requests to
-  stderr. """
-  def log_request(self, code='-', size='-'):
-    pass
+    """A SimpleHTTPRequestHandler that does not write incoming requests to
+    stderr."""
+
+    def log_request(
+        self, code: Union[int, str] = "-", size: Union[int, str] = "-"
+    ) -> None:
+        pass
+
 
 # NOTE: On Windows/Python2 tests that use this simple_server.py in a
 # subprocesses hang after a certain amount of requests (~68), if a PIPE is
 # passed as Popen's stderr argument. This problem doesn't emerge if
 # we silence the HTTP messages.
 # If you decide to receive the HTTP messages, then this bug
 # could reappear.
-use_quiet_http_request_handler = True
 
-if len(sys.argv) > 2:
-  use_quiet_http_request_handler = sys.argv[2]
+# pylint: disable=invalid-name
+handler: Type[
+    Union[SimpleHTTPRequestHandler, QuietHTTPRequestHandler]
+] = QuietHTTPRequestHandler
 
-if use_quiet_http_request_handler:
-  handler = QuietHTTPRequestHandler
-else:
-  handler = SimpleHTTPRequestHandler
+if len(sys.argv) > 2:
+    handler = SimpleHTTPRequestHandler
```

Comment:
For the purpose of this script the type annotation really seems like an overkill that impairs readability. Would it be problematic if we just did:
```python
if len(sys.argv) > 2 and sys.argv[2]:
  handler = QuietHTTPRequestHandler
else:
  handler = SimpleHTTPRequestHandler
```

Reply:
That way I receive the following two errors by `mypy`:
```
tests/simple_server.py:61: error: Cannot assign multiple types to name "handler"  without an explicit "Type[...]" annotation  [misc]

tests/simple_server.py:61: error: Incompatible types in assignment (expression has type "Type[SimpleHTTPRequestHandler]", variable has type "Type[QuietHTTPRequestHandler]")  [assignment]

Reply:
I guess you could "declare" `handler` first. Looks pretty unpythonic to me, but a bit nicer then the long line above. 
```python
handler: Type[Union[SimpleHTTPRequestHandler, QuietHTTPRequestHandler]]

if len(sys.argv) > 2 and sys.argv[2]:
  handler = QuietHTTPRequestHandler
else:
  handler = SimpleHTTPRequestHandler
```

But I'm fine with leaving it as is.

Reply:
You are right that's better even if it's not the most pythonic way it's still more understandable.
Updated the code with that suggestion.